packages/api-page-builder-aco/src/page/crud/getFolderHierarchyByPageId.ts

```diff
@@ -0,0 +1,30 @@
+import get from "lodash/get";
+import WebinyError from "@webiny/error";
+
+import { PbAcoContext } from "~/types";
+
+export const getFolderHierarchyByPageId = async (context: PbAcoContext, pageId: string) => {
+    try {
+        const { aco, pageBuilder } = context;
+
+        const page = await pageBuilder.getPage(pageId);
```

Comment:
This can be optimized by passing the whole `Page` instead of `pageId`, because in lifecycle events, you already have the entire page, so you don't need to load it again.